### PR TITLE
Fixes #15086 - window load inside ready state will not be triggered

### LINKED ISSUE #15086: jQuery 3 - window load inside ready state will not be triggered
In #14620 I proposed to put code from POS_LOAD section to jQuery ready state. But this not working in in jQuery 3. See https://github.com/jquery/jquery/issues/3194

Need to replace `jQuery(function ($) { $(window).on('load', function () { /*code*/ }); });` with `jQuery(window).on('load', function () { /*code*/ });`

Sorry for this.

### What steps will reproduce the problem?

Register some code in POS_LOAD section.

### What is the expected result?

Code must be executed when page content fully loaded.

### What do you get instead?

Code doesn't run at all.

### Additional info

| Q                | A
| ---------------- | ---
| Yii version      | 2.0.13
| PHP version      | *
| Operating system | *

### PR DESCRIPTION
**Pull request by alexantr**
| Q             | A
| ------------- | ---
| Is bugfix?    | yes
| New feature?  | no
| Breaks BC?    | no
| Tests pass?   | yes/no (for 7.1 stopped)
| Fixed issues  | #15086

### CODE REVIEW

File: framework/web/View.php [hunk 1]
```diff
@@ -604,7 +604,7 @@ protected function renderBodyEndHtml($ajaxMode)
                 $lines[] = Html::script($js, ['type' => 'text/javascript']);
             }
             if (!empty($this->js[self::POS_LOAD])) {
-                $js = "jQuery(function ($) {\n$(window).on('load', function () {\n" . implode("\n", $this->js[self::POS_LOAD]) . "\n});\n});";
+                $js = "jQuery(window).on('load', function () {\n" . implode("\n", $this->js[self::POS_LOAD]) . "\n});";
```
Comment (samdark) [inline framework/web/View.php]: Hmm... Shouldn't it be just `"jQuery(function($) {\n" . implode("\n", $this->js[self::POS_LOAD]) . "\n});"`?

Comment (alexantr) [reply framework/web/View.php]: jQuery(function($) { }); - when DOM is ready
jQuery(window).on('load', function() { }); - when content is fully loaded

Comment (samdark) [reply framework/web/View.php]: Ah, OK.

**Review by samdark — APPROVED** [review]
Merge master in, add changelog and it's done.

File: framework/web/View.php (hunk 1 above)
Comment (cebe) [reply framework/web/View.php]: I see that we are nesting the calls unnecessarily here but as far as I understand DOM ready event comes before window load event. if we register the handler inside of the DOM ready event, why does it have no effect?

Comment (alexantr) [reply framework/web/View.php]: In version 3 the ready handler fires asynchronously.

You can check code:

```javascript
jQuery(function ($) {
  console.log('ready');
  $(window).on('load', function () {
    console.log('load in ready');
  });
});
jQuery(window).on('load', function () {
  console.log('load');
});
```

With jQuery 2.2.4 you will see:

```
ready
load
load in ready
```

With jQuery 3.2.1 you will see:

```
load
ready
```

No "load in ready" anymore.

https://github.com/jquery/jquery/issues/3194#issuecomment-228556922
https://github.com/jquery/jquery/issues/3197

Comment (alexantr) [reply framework/web/View.php]: I just wanted to make `$` working even jQuery.noConflict() is enabled. But missed this great change in jQuery API. The best way to fix this bug - rollback my proposal in #14620

**Review by SilverFire — APPROVED** [review]

**Comment by SilverFire** [comment]
Merged: https://github.com/yiisoft/yii2/commit/22467864834b78236469882e6bd169144995161e

**Comment by SilverFire** [comment]
Thank you!

**Pull request closed without merging**